dashboard/app/controllers/xhr_proxy_controller.rb

```diff
@@ -0,0 +1,48 @@
+# A controller for proxying web requests to 3rd party APIs. This protects
+# applab users from XSS attacks, complies with the Same Origin Policy,
+# and prevents http/https mismatch warnings in IE.
+#
+# Responses are cached for one minute since many 3rd party APIs serve data
+# which changes frequently such as news or sports scores.
+#
+# To reduce the likelihood of abuse, we only proxy content with an allowed
+# whitelist of media types. We will need to monitor usage to detect
+# abuse and potentially add other abuse prevention measures.
+
+require 'set'
+
+class XhrProxyController < ApplicationController
+  include ProxyHelper
+
+  ALLOWED_CONTENT_TYPES = Set.new(['application/json', 'text/json', 'text/javascript'])
+
+  # How long the content is allowed to be cached
+  EXPIRY_TIME = 1.minute
+
+  # Return the proxied api at the given URL.
+  def get
+    channel_id = params[:c]
+    url = params[:u]
+
+    begin
+      owner_storage_id, _ = storage_decrypt_channel_id(channel_id)
+    rescue Exception => e
```

Comment:
Is it possible to be more specific about the type of exception you catch to avoid masking other errors?